sklearn/ensemble/_gb.py

```diff
@@ -265,21 +266,27 @@ def _fit_stage(
 
     def _check_params(self):
         """Check validity of parameters and raise ValueError if not valid."""
-        if self.n_estimators <= 0:
-            raise ValueError(
-                "n_estimators must be greater than 0 but was %r" % self.n_estimators
-            )
 
-        if self.learning_rate <= 0.0:
-            raise ValueError(
-                "learning_rate must be greater than 0 but was %r" % self.learning_rate
-            )
+        check_scalar(
+            self.learning_rate,
+            name="learning_rate",
+            target_type=numbers.Real,
+            min_val=0.0,
+            include_boundaries="neither",
+        )
+
+        check_scalar(
+            self.n_estimators,
+            name="n_estimators",
+            target_type=numbers.Integral,
+            min_val=1,
```

Comment:
Even if this is correct, we could be explicit. It will remove the need to check what is the default behaviour of `check_scalar`.

```suggestion
            min_val=1,
            include_boundaries="left",
```

Reply:
Yes, that's a great point